presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcUpdatablePageSource.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor.storage;
+
+import com.facebook.presto.orc.OrcBatchRecordReader;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.UpdatablePageSource;
+import com.facebook.presto.spi.block.Block;
+import io.airlift.slice.Slice;
+
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+
+public class OrcUpdatablePageSource
+        implements UpdatablePageSource
+{
+    private final Optional<ShardRewriter> shardRewriter;
+    private final BitSet rowsToDelete;
+
+    private OrcPageSource parentPageSource;
+
+    public OrcUpdatablePageSource(
+            Optional<ShardRewriter> shardRewriter,
+            OrcBatchRecordReader recordReader,
+            OrcPageSource parentPageSource)
+    {
+        this.rowsToDelete = new BitSet(toIntExact(recordReader.getFileRowCount()));
```

Comment:
move the assignment after `shardRewriter` to align with the on-stack declaration. Also requireNonNull on `recordReader`